dist2src/worker/updater.py

```diff
@@ -53,67 +52,65 @@ def check_updates(
         else:
             logger.debug("Celery tasks created never expire")
 
-        m = re.match(
-            r"(?P<fork>forks\/)?((?P<owner>.+)\/)?(?P<namespace>.+)",
-            self.cfg.src_git_namespace,
+        src_gitlab_group = self.cfg.src_git_svc.gitlab_instance.groups.get(
+            self.cfg.src_git_namespace
         )
-
-        # Get repositories in source-git
-        url = f"{self.cfg.src_git_svc.api_url}projects"
-        params = {
-            "namespace": m["namespace"],
-            "pattern": project,
-            "fork": m["fork"] is not None,
-            "per_page": self.PROJECTS_PER_PAGE,
-            "owner": m["owner"] or None,
-            "short": True,
-        }
-        while url:
-            r = self.cfg.src_git_svc.call_api(url, params=params)
-            # The URL in 'next' already has the required parameters
-            url, params = r["pagination"]["next"], None
-            logger.debug(
-                f"Next page: {url!r}. Total pages: {r['pagination']['pages']!r}."
+        if project:
+            # update only the selected project
+            gitlab_project = self.cfg.src_git_svc.get_project(
+                repo=project, namespace=self.cfg.src_git_namespace
             )
-            for src_git_project in r["projects"]:
-                try:
-                    self._check_project(src_git_project["name"], branch)
-                except (OgrException, RetryError):
-                    logger.exception(
-                        f"Failed checking project {src_git_project['name']!r}"
+            self.check_and_update_project(gitlab_project, branch=branch)
+        else:
+            for n in range(1, self.PROJECTS_PER_PAGE):
```

Comment:
Hm, I don't really understand how this is supposed to work. :thinking: 

I would expect that `projects.list()` bellow gets called with `per_page=self.PROJECTS_PER_PAGE`, while `n` is incremented, and the iteration stops when there are no projects returned.

But the current code tells me a different story.

Reply:
oh yeah, they are swapped; will fix that and provide a comment